File: 第七章_二叉树/二叉树(4)/折纸练习题/FoldPaper.cpp
class FoldPaper {
public:
    vector<string> foldPaper(int n) {
        vector<string> res;
        Inorder(res,n,"down");
        return res;
    }
 
    void Inorder(vector<string> &res,int n,string val)
    {
        if(n==0)
        {
            return ;
        }
         
        Inorder(res,n-1,"down");
        res.push_back(val);
        Inorder(res,n-1,"up");
    }
};

File: 第七章_二叉树/二叉树(4)/折纸练习题/FoldPaper.java
import java.util.ArrayList;

public class FoldPaper {
	public String[] foldPaper(int n) {
		ArrayList<String> list = new ArrayList<>();
		Inorder(list, n, "down");
		String[] arr = new String[list.size()];
		for (int i = 0; i < list.size(); i++) {
			arr[i] = list.get(i);
		}
		return arr;
	}

	// ÀàËÆÖÐÐò±éÀú
	public void Inorder(ArrayList<String> list, int n, String val) {
		if (n == 0) {
			return;
		}
		Inorder(list, n - 1, "down");
		list.add(val);
		Inorder(list, n - 1, "up");
	}
}